transfer/adapterbase.go

```diff
@@ -103,8 +106,15 @@ func (a *adapterBase) worker(workerNum int) {
 			}
 		}
 		tracerx.Printf("xfer: adapter %q worker %d processing job for %q", a.Name(), workerNum, t.Object.Oid)
+
 		// Actual transfer happens here
-		err := a.transferImpl.DoTransfer(t, a.cb, authCallback)
+		var err error
+		if t.Object.IsExpired(time.Now()) {
```

Comment:
As suggested earlier I would add a few seconds grace to `time.Now()` before passing it to `IsExpired` to avoid the case where the link is not expired exactly when you test, but is by the time the request goes through.